pkg/rpc/client/client.go

```diff
@@ -81,10 +81,20 @@ type calculateValidUntilBlockCache struct {
 // New returns a new Client ready to use. You should call Init method to
 // initialize network magic the client is operating on.
 func New(ctx context.Context, endpoint string, opts Options) (*Client, error) {
-	url, err := url.Parse(endpoint)
+	cl, err := newClient(ctx, endpoint, opts)
 	if err != nil {
 		return nil, err
 	}
+	return &cl, nil
+}
+
+func newClient(ctx context.Context, endpoint string, opts Options) (Client, error) {
```

Comment:
`govet` still complains about mutex being copied. What about `initClient(&cl, endpoint, opts)`? Can use it with `&ws.Client`.